lib/record.gd

```diff
@@ -0,0 +1,198 @@
+#############################################################################
+##
+#W  record.gd                    GAP library                     Thomas Breuer
+#W                                                             & Frank Celler
+##
+##
+#Y  Copyright (C)  1997,  Lehrstuhl D für Mathematik,  RWTH Aachen,  Germany
+#Y  (C) 1998 School Math and Comp. Sci., University of St Andrews, Scotland
+#Y  Copyright (C) 2002 The GAP Group
+##
+##  This file contains methods for records.
+##  Compared to &GAP; 3, where records were used to represent domains and
+##  all kinds of external arithmetic objects, in &GAP; 4 there is no
+##  important role for records.
+##  So the standard library provides only methods for `PrintObj', `String',
+##  `=', and `<', and the latter two are not installed to compare records
+##  with objects in other families.
+##
+##  In order to achieve a special behaviour of records as in &GAP; 3 such
+##  that a record can be regarded as equal to objects in other families
+##  or such that a record can be compared via `<' with objects in other
+##  families, one can load the file `compat3c.g'.
```

Comment:
Might as well remove the reference to `compat3c.g`.

Also, I think this renaming and splitting of `record.g` is of independent interest, so we could merge it before the rest of this PR (which would reduce the diff in this PR, making it easier to review).